src/commands/org/create/sandbox.ts

```diff
@@ -155,9 +173,13 @@ export default class CreateSandbox extends SandboxCommandBase<SandboxCommandResp
       );
       delete sandboxReq.ActivationUserGroupName;
     }
+
     return {
       ...sandboxReq,
-      ...(this.flags.clone ? { SourceId: await this.getSourceId() } : {}),
+      ...(srcSandboxName ?? this.flags['source-sandbox-name']
+        ? { SourceId: await requestFunctions.getSrcIdByName(this.flags['target-org'].getConnection(), srcSandboxName) }
+        : {}),
+      ...(srcId ?? this.flags['source-id'] ? { SourceId: srcId } : {}),
```

Comment:
ditto, `srcId` should have the flag value, def. file value or `undefined`
https://github.com/salesforcecli/plugin-org/pull/1237/files#r1805117017